### PR TITLE
Enhance endpoints handling

### PR DESCRIPTION
**Pull request by ask4gilles**
Provide spring boot actuator skip patterns dynamically.

### CODE REVIEW

**Comment by marcingrzejszczak** [comment]
Are you sure you've commited all the files? :Pline

**Comment by ask4gilles** [comment]
Apparently not... it's a teaser :) I'll try to push the rest ASAP.

**Comment by ask4gilles** [comment]
Should be ok now.

**Comment by marcingrzejszczak** [comment]
@shakuzen can you check this out please?

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/SleuthWebProperties.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 	/**
 	 * Default set of skip patterns.
 	 */
-	public static final String DEFAULT_SKIP_PATTERN = "/api-docs.*|/autoconfig|/configprops|/dump|/health|/info|/metrics.*|/mappings|/trace|/swagger.*|.*\\.png|.*\\.css|.*\\.js|.*\\.html|/favicon.ico|/hystrix.stream|/application/.*|/actuator.*|/cloudfoundryapplication";
+	public static final String DEFAULT_SKIP_PATTERN = "/api-docs.*|/swagger.*|.*\\.png|.*\\.css|.*\\.js|.*\\.html|/favicon.ico|/hystrix.stream|/application/.*|/cloudfoundryapplication";
```
[the remark this answers is not in the document]
Comment (ask4gilles) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/SleuthWebProperties.java]: No idea, @marcingrzejszczak ?

[the remark this answers is not in the document]
Comment (ask4gilles) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/SleuthWebProperties.java]: No idea, @marcingrzejszczak ?

**Comment by ask4gilles** [comment]
@shakuzen The skip pattern for the actuator endpoints now looks like this:
`(archaius|archaius/.*|auditevents|auditevents/.*|beans|beans/.*|caches|caches/.*|health|health/.*|conditions|conditions/.*|configprops|configprops/.*|env|env/.*|info|info/.*|integrationgraph|integrationgraph/.*|loggers|loggers/.*|heapdump|heapdump/.*|threaddump|threaddump/.*|metrics|metrics/.*|scheduledtasks|scheduledtasks/.*|httptrace|httptrace/.*|mappings|mappings/.*|refresh|refresh/.*|features|features/.*)`
I did not find a better way to handle it. I've added different IT tests for the different cases you mentioned (I think). This can probably be improved...
@adriancole Can you please have a look :) ?

**Comment by ask4gilles** [comment]
Hi @marcingrzejszczak , @adriancole , @shakuzen ,
Regarding the last comments, I've introduced the following flag in my last commit:
`spring.sleuth.web.ignoreAutoConfiguredSkipPatterns`

 ```
/**
 * If set to true, auto-configured skip patterns will be ignored.
* @see TraceWebAutoConfiguration
*/
private boolean ignoreAutoConfiguredSkipPatterns = false;
 ```
If this flag is false, actuator and management skip patterns are auto-configured.
Otherwise, only the default `skipPattern` and the `additionalSkipPattern` are taken into account.
I hope that this meets your requirements...

**Comment by marcingrzejszczak** [comment]
We're almost done, everything looks great so far. Can you update the docs too?

**Comment by ask4gilles** [comment]
> We're almost done, everything looks great so far. Can you update the docs too?

See last commit ;)

**Comment by marcingrzejszczak** [comment]
Congrats @ask4gilles on this awesome fix!

**Comment by ask4gilles** [comment]
@marcingrzejszczak I think you can close the related issue?

**Pull request merged**